src/middlewares/cors.rs

```diff
@@ -0,0 +1,58 @@
+use std::pin::Pin;
+
+use futures::{FutureExt, TryFutureExt, future};
+use gotham::handler::HandlerFuture;
+use gotham::hyper::header::{HeaderMap, ORIGIN};
+use gotham::middleware::Middleware;
+use gotham::prelude::*;
+use gotham::state::State;
+
+use std::env;
+
+#[derive(Clone, NewMiddleware)]
+pub struct CorsMiddleware;
+
+impl Middleware for CorsMiddleware {
+    fn call<Chain>(self, state: State, chain: Chain) -> Pin<Box<HandlerFuture>>
+    where
+        Chain: FnOnce(State) -> Pin<Box<HandlerFuture>>,
+    {
+        dotenv::dotenv().ok();
```

Comment:
Is this dotenv needed here? I suggest to move this one to main instead of calling from middlewares

Reply:
I tried moving it to the main and it works properly. Good catch!